datastream-server/src/test/java/com/linkedin/datastream/server/TestLoadBasedTaskCountEstimator.java

```diff
@@ -0,0 +1,85 @@
+/**
+ *  Copyright 2021 LinkedIn Corporation. All rights reserved.
+ *  Licensed under the BSD 2-Clause License. See the LICENSE file in the project root for license information.
+ *  See the NOTICE file in the project root for additional information regarding copyright ownership.
+ */
+package com.linkedin.datastream.server;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import com.linkedin.datastream.server.assignment.LoadBasedTaskCountEstimator;
+import com.linkedin.datastream.server.providers.FileBasedPartitionThroughputProvider;
+
+
+/**
+ * Tests for {@link LoadBasedTaskCountEstimator}
+ */
+public class TestLoadBasedTaskCountEstimator {
+  private static final String THROUGHPUT_FILE_NAME = "partitionThroughput.json";
+  FileBasedPartitionThroughputProvider _provider;
```

Comment:
add negative tests for throughput information not present, assigned partitions, unassigned partitions